src/skalo/utils.rs

```diff
@@ -0,0 +1,298 @@
+use hashbrown::HashMap;
+use std::path::PathBuf;
+use std::sync::OnceLock;
+
+
+// structure to hold arguments
+#[derive(Debug)]
+pub struct Config {
+    pub input_file: String,
+    pub output_name: String,
+    pub max_missing: f32,
+    pub max_depth: usize,
+    pub	max_indel_kmers: usize,
+    pub nb_threads: usize,
+    pub reference_genome: Option<PathBuf>,
+}
+pub static CONFIG: OnceLock<Config> = OnceLock::new();
+
+
+// structure to hold dataset information
+#[derive(Debug, Clone)]
+pub struct DataInfo {
+    pub k_graph: usize,
+    pub sample_names: Vec<String>,
+}
+pub static DATA_INFO: OnceLock<DataInfo> = OnceLock::new();
+
+
+
+
+/*
+pub fn resolve_upac(nucl1: &str, nucl2: &str) -> String {
+    match (nucl1, nucl2) {
+        // Basic nucleotide conflicts
+        ("A", "G") | ("G", "A") => "R".to_string(),
+        ("C", "T") | ("T", "C") => "Y".to_string(),
+        ("G", "C") | ("C", "G") => "S".to_string(),
+        ("A", "T") | ("T", "A") => "W".to_string(),
+        ("G", "T") | ("T", "G") => "K".to_string(),
+        ("A", "C") | ("C", "A") => "M".to_string(),
+
+        // Handling conflicts with existing UPAC codes
+        ("A", "R") | ("R", "A") | ("G", "R") | ("R", "G") => "R".to_string(),
+        ("C", "Y") | ("Y", "C") | ("T", "Y") | ("Y", "T") => "Y".to_string(),
+        ("G", "S") | ("S", "G") | ("C", "S") | ("S", "C") => "S".to_string(),
+        ("A", "W") | ("W", "A") | ("T", "W") | ("W", "T") => "W".to_string(),
+        ("G", "K") | ("K", "G") | ("T", "K") | ("K", "T") => "K".to_string(),
+        ("A", "M") | ("M", "A") | ("C", "M") | ("M", "C") => "M".to_string(),
+
+        // Mixed UPAC code resolution with specific nucleotides
+        ("A", "Y") | ("Y", "A") => "V".to_string(),
+        ("G", "Y") | ("Y", "G") => "B".to_string(),
+        ("T", "R") | ("R", "T") => "D".to_string(),
+        ("C", "R") | ("R", "C") => "V".to_string(),
+
+        // Extended UPAC combinations for three nucleotides
+        ("A", "V") | ("V", "A") | ("C", "V") | ("V", "C") | ("G", "V") | ("V", "G") => "V".to_string(),
+        ("A", "H") | ("H", "A") | ("C", "H") | ("H", "C") | ("T", "H") | ("H", "T") => "H".to_string(),
+        ("A", "D") | ("D", "A") | ("G", "D") | ("D", "G") | ("T", "D") | ("D", "T") => "D".to_string(),
+        ("C", "B") | ("B", "C") | ("G", "B") | ("B", "G") | ("T", "B") | ("B", "T") => "B".to_string(),
+
+        // Default to N for any unresolved conflicts
+        _ => "N".to_string(),
+    }
+}
+*/
+
+
+pub fn rev_compl(seq: &str) -> String {
```

Comment:
All the rest of these functions should probably go in bit_encoding.rs, and check they aren't duplicated